### PR TITLE
Python 3 compatibility

### PR DESCRIPTION
**Pull request by dirk-thomas**

### CODE REVIEW

**Comment by dirk-thomas** [comment]
@wjwwood please review.

File: tools/rosbag/src/rosbag/bag.py [hunk 1]
```diff
@@ -1274,12 +1287,39 @@ def __str__(self):
...
+        return self._compare(other, lambda s, o: s != o)
+
+class _IndexEntry(ComparableMixin):
     def __init__(self, time):
         self.time = time
 
-    def __cmp__(self, other):
-        return self.time.__cmp__(other.time)
+    def _cmpkey(self):
+        return self.time
```
Comment (wjwwood) [inline tools/rosbag/src/rosbag/bag.py:1322]: I had never seen this before, but after looking into it, it seems reasonable.

**Comment by wjwwood** [comment]
+1

File: tools/rosbag/src/rosbag/bag.py (hunk 1 above)
Comment (dirk-thomas) [inline tools/rosbag/src/rosbag/bag.py:1322]: :wink:

**Pull request merged**